### PR TITLE
Bugfix/ls24004056/packed coverage

### PR DESCRIPTION
**Pull request by dom-apuliasoft**
## Description

This PR does not add or change any RPG language functionality

From a technical standpoint changes are:
- Addition of new test cases for PACKED value and the interoperability of DS and strings
- The removal of a check added in #600 that caused a regression in number coercion

Related to:
- LS24004056
- #600 
- #602 
- #617 

## Checklist:
- [x] If this feature involves RPGLE fixes or improvements, they are well-described in the summary.
- [x] There are tests for this feature.
- [x] RPGLE code used for tests is easily understandable and includes comments that clarify the purpose of this feature.
- [x] The code follows Kotlin conventions (run `./gradlew ktlintCheck`).
- [x] The code passes all tests (run `./gradlew check`).
- [ ] Relevant documentation is included in the `docs` directory.

### CODE REVIEW

**Review by lanarimarco — CHANGES_REQUESTED** [review]
I appreciate implementations where you cut than adding to it! 😊

In addition to my comments on specific files, it seems there's no test that follows this structure:

- Declare a `DS1` with `StringType`, `NumberType` zoned, and `NumberType` packed.
- Declare `S1`, a standalone `StringType` similar to `DS1`.
- Declare `DS2`, structured like `DS1`.
- Assign `DS1` to `S1` and then assign `S1` to `DS2`.
- Assert that each field of `DS1` is equal to the corresponding field in `DS2`.

File: rpgJavaInterpreter-core/src/test/resources/mute/MUTE13_41.rpgle [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+     DJALOGT2        E DS                  EXTNAME(JALOGT0F) INZ
+     DSJALOG           S                   LIKE(JALOGT)
+      *
+      /COPY QILEGEN,£TABB£1DS
+      /COPY QILEGEN,£PDS
+      *---------------------------------------------------------------------
+    RD* MAIN
+      *---------------------------------------------------------------------
+      *
+     C                   EVAL      J1DTIN=20230930
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/test/resources/mute/MUTE13_41.rpgle:18]: Why not `JALOGT.J1DTIN` for this fields and for all fields below?

File: rpgJavaInterpreter-core/src/test/resources/mute/MUTE13_41.rpgle [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+     C                   EVAL      J1ORIN=113123
+     C                   EVAL      J1DTUL=20230930
+     C                   EVAL      J1ORUL=113233
+     C                   EVAL      J1DTFI=20230930
+     C                   EVAL      J1ORFI=113312
+     C                   EVAL      SJALOG=JALOGT
+    MU* VAL1(JALOGT) VAL2(JALOGT2) COMP(EQ)
+     C                   EVAL      JALOGT=SJALOG
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/test/resources/mute/MUTE13_41.rpgle:26]: Why not `JALOGT2=SJALOG`

**Review by lanarimarco — COMMENTED** [review]
I have a doubt about a test

File: rpgJavaInterpreter-core/src/test/resources/mute/MUTE13_41.rpgle [hunk 3]
```diff
@@ -0,0 +1,29 @@
+     V*=====================================================================
+     V* MODIFICHE Ril.  T Au Descrizione
+     V* gg/mm/aa  nn.mm i xx Breve descrizione
+     V*=====================================================================
+     V* 24/09/24  V6R1   BMA Creato
+     V*=====================================================================
+     FJALOGT0F  UF A E             DISK
+     DJALOGT         E DS                  EXTNAME(JALOGT0F) INZ
+     DJALOGT2        E DS                  EXTNAME(JALOGT0F) INZ
```
Comment (benetti-smeup) [inline rpgJavaInterpreter-core/src/test/resources/mute/MUTE13_41.rpgle:9]: @dom-apuliasoft It’s not possible to define two data structures based on a file without renaming the fields. The fields become duplicated, and the program cannot be compiled on IBMi. You changed the example without testing that it works on IBMi.

**Comment by dom-apuliasoft** [comment]
As I understand my changes to `MUTE13_41` were not appropriate I rolled it back to the original version @benetti-smeup gave me and only added the MUTE assertions in it. Let me know if it ok now and else-wise what needs to be changed.

**Review by lanarimarco — APPROVED** [review]

**Pull request merged**